src/dwarf/Gparser.c

```diff
@@ -1022,6 +1022,9 @@ find_reg_state (struct dwarf_cursor *c, dwarf_state_record_t *sr)
     }
   else
     {
+      if (cache)
+        put_rs_cache (c->as, cache, &saved_mask);
```

Comment:
Hi @djwatson, we encountered a tricky issue and can certainly use your help here. Holding the cache lock while calling into the dynamic loader below could cause deadlocks (since DL may trigger unwind as well). Is dropping and reacquiring the cache the right fix here?